plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2124,31 +2135,61 @@ public ConnectorAnalyzeMetadata getStatisticsCollectionMetadata(ConnectorSession
                 handle.getReadVersion(),
                 false);
 
+        TableStatisticsMetadata statisticsMetadata = getStatisticsCollectionMetadata(
+                statistics,
+                extractColumnMetadata(metadata, typeManager),
+                analyzeColumnNames.orElse(allColumnNames),
+                true);
+
+        return new ConnectorAnalyzeMetadata(newHandle, statisticsMetadata);
+    }
+
+    @Override
+    public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        if (!isCollectExtendedStatisticsColumnStatisticsOnWrite(session)) {
+            return TableStatisticsMetadata.empty();
+        }
+
+        Set<String> allColumnNames = tableMetadata.getColumns().stream()
+                .map(ColumnMetadata::getName)
+                .collect(toImmutableSet());
+
+        return getStatisticsCollectionMetadata(
+                Optional.empty(),
+                tableMetadata.getColumns(),
+                allColumnNames,
+                false);
```

Comment:
Why not `includeMaxFileModifiedTime` in this situation?

Reply:
Statistic aggregation during table creation does not have information about file_modified_time yet.

Reply:
Right, right. Then if the modified time isn't present we just use the current time when the collection is done. Makes sense.

Reply:
Can you please add a code comment explaining this consideration?
What do we need to have this information available?